docs/source/architecture/shared_objects.rst

```diff
@@ -0,0 +1,133 @@
+Iroha Shared Objects Description
+================================
+
+This document describes access to shared objects in a multithreaded scenario, whether or not is properly synchronized.
+
+Document Structure
+------------------
+
+Each component in this document has its own diagram describing all members, and whether or not access to them is synchronized or not:
+
+* **Synchronized**
```

Comment:
The cell background color manipulation was used for highlighting in the original doc.
I believe that making the font bold would not bring enough distinction to viewers.

Reply:
Done. Please check if this is ok?